qiskit/circuit/library/n_local/n_local.py

```diff
@@ -948,11 +919,11 @@ def _build(self) -> None:
         circuit = QuantumCircuit(*self.qregs, name=self.name)
 
         # use the initial state as starting circuit, if it is set
-        if self._initial_state:
-            if isinstance(self._initial_state, QuantumCircuit):
-                initial = self._initial_state.copy()
+        if self.initial_state:
+            if isinstance(self.initial_state, QuantumCircuit):
+                initial = self.initial_state.copy()
             else:
-                initial = self._initial_state.construct_circuit("circuit", register=self.qregs[0])
+                initial = self.initial_state.construct_circuit("circuit", register=self.qregs[0])
```

Comment:
Good point, it set's `self._initial_state_circuit` to a circuit we can use this instead.

Reply:
Actually, after looking at it more closely, some circuits (like QAOA) depend on constructing the initial state at build-time so we can't pre-build the initial state circuit. Since the whole `initial_state.construct_circuit` logic can soon be removed (was deprecated with 0.18) I'd prefer to leave it as is and just delete it with the next release.